mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,623 @@
+        `[pac_out, low_freq_phase, hi_freq_amplitude]`.
+    concat_epochs : bool
+        If True, epochs will be concatenated before calculating PAC values. If
+        epochs are relatively short, this is a good idea in order to improve
+        stability of the PAC metric.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see `mne.verbose`).
+
+    Returns
+    -------
+    pac_out : array, list of arrays, dtype float, shape (n_epochs, n_pairs,
+              n_times)
+        The computed phase-amplitude coupling between
+        each pair of data sources given in ixs. If multiple pac metrics are
+        specified, there will be one array per metric in the output list.
+    [phase_signal] : array, shape (n_phase_signals, n_times,)
+        Only returned if `return_data` is True. The phase timeseries of the
+        phase signals (first column of `ixs`).
+    [amp_signal] : array, shape (n_amp_signals, n_times,)
+        Only returned if `return_data` is True. The amplitude timeseries of the
+        amplitude signals (second column of `ixs`).
+
+    References
+    ----------
+    [1] This function uses the PacPy module developed by the Voytek lab.
+        https://github.com/voytekresearch/pacpy
+    """
+    from mne.io.base import _BaseRaw
+    if not isinstance(inst, _BaseRaw):
+        raise ValueError('Must supply Raw as input')
+    sfreq = inst.info['sfreq']
+    data = inst[:][0]
+    pac = _phase_amplitude_coupling(data, sfreq, f_phase, f_amp, ixs,
+                                    pac_func=pac_func, events=events,
+                                    tmin=tmin, tmax=tmax, n_cycles=n_cycles,
+                                    scale_amp_func=scale_amp_func,
+                                    return_data=return_data,
+                                    concat_epochs=concat_epochs,
+                                    n_jobs=n_jobs, verbose=verbose)
+    # Collect the data properly
+    if return_data is True:
+        pac, freq_pac, data_ph, data_am = pac
+        return pac, freq_pac, data_ph, data_am
+    else:
+        pac, freq_pac = pac
+        return pac, freq_pac
+
+
+def _phase_amplitude_coupling(data, sfreq, f_phase, f_amp, ixs,
+                              pac_func='ozkurt', events=None,
+                              tmin=None, tmax=None, n_cycles=3,
+                              scale_amp_func=None, return_data=False,
+                              concat_epochs=False, n_jobs=1,
+                              verbose=None):
+    """ Compute phase-amplitude coupling using pacpy.
+
+    Parameters
+    ----------
+    data : array, shape ([n_epochs], n_channels, n_times)
+        The data used to calculate PAC
+    sfreq : float
+        The sampling frequency of the data.
+    f_phase : array, dtype float, shape (n_bands, 2,)
```

Comment:
what is `n_bands`?
I think it does not need to be the same as in `f_amp.shape[0]`.
Is it the same as `n_phase_signals` below?